boot/boot.go

```diff
@@ -327,3 +327,33 @@ func MarkBootSuccessful(dev Device) error {
 	}
 	return nil
 }
+
+var ErrUnsupportedSystemBootMode = errors.New("system boot mode is unsupported")
+
+// SetRecoveryBootSystemAndMode configures the bootloader to boot into the given
+// recovery system in a particular mode. When booting into a recovery system is
+// not supproted by the device, an ErrUnsupportedSystemBootMode error is
+// returned.
+func SetRecoveryBootSystemAndMode(dev Device, systemLabel, mode string) error {
+	if !dev.HasModeenv() {
+		// only UC20 devices are supported
+		return ErrUnsupportedSystemBootMode
+	}
+
+	ferr := func(err error) error {
+		return fmt.Errorf("cannot set up the boot environment: %v", err)
+	}
+
+	s := &bootSystemState20{}
```

Comment:
can't the code attach to the struct just be inlined here?

Reply:
Ah, you mean drop the `bootsystemstate20` wrapper. I'll push it next.